open-class/docs/classes/assignment-branch-logic.md

```diff
@@ -10,244 +10,187 @@ import useBaseUrl from '@docusaurus/useBaseUrl';
 
-1. Short
-1. Descriptive (without cryptic abbreviations)
-1. Lower-case
-1. Use _ to separate multiple words when needed
+### If this, then that
 
-We are going to need true/false variables and an ending screen. Let's name them as follows:
+We'll use what's called an `if` statement. It tells our program, 'if the situation is this, then do that.'
 
-variable name | meaning
---------------|---------
-go_to_party | A screen that tells the person to go to the party
-dont_go_to_party | A screen that says not to go to the party
-likes_parties | Do you like parties?
-wants_to_meet | Do you want to meet someone there?
-big_assignment | Do you have a big assignment due?
-
-### Writing our first question blocks
-
-Let's start with writing the ending blocks, which are the simplest. Remember, we will use the `event` specifier for these because they are ending screens.
-
-```yaml
+``` yaml
+---
+id: flow
+mandatory: True
+code: |
+  if ( user_likes_parties ):
+    go_to_party
+  else:
+    dont_go_to_party
 ---
-event: go_to_party
-question: |
-  You should go to the party!
-subquestion: |
-  [FILE https://media.giphy.com/media/l0MYt5jPR6QX5pnqM/giphy.gif]
 ```
 
-[![](https://media.giphy.com/media/l0MYt5jPR6QX5pnqM/giphy.gif)]
+### More branches!
+
+So that's not really all that matters, right? For example, do they have a ride or would they have to walk?
 
 ```yaml
 ---
-event: dont_go_to_party
+id: has_a_ride
 question: |
-  You should just go to sleep
-subquestion: |
-  [FILE https://media.giphy.com/media/xT8qBvH1pAhtfSx52U/giphy.gif]
+  Do you have a ride?
+fields:
+  - yesno: user_has_a_ride
+---
 ```
 
-[![](https://media.giphy.com/media/xT8qBvH1pAhtfSx52U/giphy.gif)]
-
-Let's try writing one of the yes/no questions.
+We could ask them both questions separately...
 
-```yaml
----
-question: |
-  Do you like parties?
-yesno: likes_parties
+``` yaml
 ---
-question: |
-  Do you want to meet someone at the party?
-yesno: wants_to_meet
+id: flow
+mandatory: True
+code: |
+  if ( user_likes_parties ):
+    go_to_party
+  else:
+    dont_go_to_party
+  
+  if ( user_has_a_ride ):
+    go_to_party
+  else:
+    dont_go_to_party
 ---
-question: | 
-  Do you have a big assignment due?
-yesno: big_assignment 
 ```
 
-### Setting up the branch logic
+...but we'd be wasting their time. If they don't like parties, why does it matter if they have a ride? They want their answer as soon as possible! So we nest the `if` statements inside each other. We only ask the second question if they said yes to the first question.
 
-We have already seen in class how we can use a `code` block to control question order. Now, we're going to combine that with `if`/`else` statements. These will follow pretty well regular English expressions.
-
-```yaml
+``` yaml
 ---
-comment: |
-  We will write the questions below in the order we want them 
-  to appear. Optional questions will be underneath an if statement.
-
-  This will be the only mandatory block in our interview
```

Comment:
Question: What have they learned about mandatory block so far?

Reply:
We went over it in class--it was something brought up as a question right away, and also something people who had used Docassemble before struggled with, so I want to reinforce it each time I use it. About 1/2 of the students got it for assignment 1, so I will use using it in all of the example code, but won't expect people to do it from scratch on their own.

Reply:
I'll try to remember that. Did you folks go over it being used in question blocks only? In code blocks only? Both?